tests/test_items.cpp

```diff
@@ -983,6 +983,57 @@ TEST_F(ItemTest, CreateItemsWithEmptyVector)
     }
     FAIL();
 }
+
+TEST_F(ItemTest, SendItemsWithVector)
+{
+    std::vector<ews::message> messages;
+    auto message = ews::message();
+    std::vector<ews::mailbox> recipients;
+    for (int i = 0; i < 10; i++)
+    {
+        message.set_subject("?");
+        recipients.push_back(ews::mailbox("whoishere@home.com"));
+        message.set_to_recipients(recipients);
+        messages.push_back(message);
+    }
+
+    auto message_ids =
+        service().create_item(messages, ews::message_disposition::save_only);
+    ews::internal::on_scope_exit remove_messages([&] {
+        for (auto& id : message_ids)
+        {
+            service().delete_item(id);
+        }
+    });
+
+    EXPECT_NO_THROW(service().send_item(message_ids));
+}
+
+TEST_F(ItemTest, SendItemsWithEmptyVector)
+{
+    std::vector<ews::message> messages;
+    auto message = ews::message();
+    std::vector<ews::mailbox> recipients;
+    for (int i = 0; i < 10; i++)
+    {
+        message.set_subject("?");
+        recipients.push_back(ews::mailbox("whoishere@home.com"));
+        message.set_to_recipients(recipients);
+        messages.push_back(message);
+    }
+
+    auto message_ids =
+        service().create_item(messages, ews::message_disposition::save_only);
+    ews::internal::on_scope_exit remove_messages([&] {
+        for (auto& id : message_ids)
+        {
+            service().delete_item(id);
+        }
+    });
+    message_ids.clear();
+
+    EXPECT_THROW(service().send_item(message_ids), ews::exception);
+}
 }
 
```

Comment:
I think you are missing a test for this overload

```cpp
void send_item(const std::vector<item_id>&, const folder_id&);
```
Any reason to not cover this?

Reply:
The overload is always being called by default.
See:
```cpp
void send_item(const std::vector<item_id>& ids)
{
    send_item(ids, folder_id());
}
```
So no need to cover it specifically.

Reply:
Only 50% correct. You only test the invalid-folder-id case here. Might be good to add a test where you add a valid folder_id so that corresponding XML gets generated and send to the server.